packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4734,11 +4734,60 @@ class Listener extends SingleChildRenderObjectWidget {
-/// previously. Similarly, when the child repaints but the surround tree does
-/// not, we can re-record its display list without re-recording the display list
-/// for the surround tree.
+/// the surrounding parts of the tree.
+///
+/// This is useful since [RenderObject.paint] may be triggered even if its
+/// associated [Widget] instances' did not change or rebuild. A [RenderObject]
+/// will repaint whenever any [RenderObject] that shares the same [Layer] is
+/// marked as being dirty and needing paint (see [RenderObject.markNeedsPaint]).
+///
+/// such as when an
+/// ancestor scrolls or when an ancestor or descendent's widget is going through
+/// an animation.
+///
+/// Containing [RenderObject.paint] to parts of the render subtree that are
+/// actually visually changing using [RepaintBoundary] explicitly or implicitly
+/// is therefore critical to minimizing redundant work and improving the app's
+/// performance. Render tree repaints can be monitored via debug flags such as
+/// [debugRepaintRainbowEnabled] or [debugProfilePaintsEnabled].
+///
+/// When a [RenderObject] is flagged as needing to paint via
+/// [RenderObject.markNeedsPaint], the nearest ancestor [RenderObject] up to
+/// possibly the root of the application with [RenderObject.isRepaintBoundary]
+/// is alerted to repaint. That nearest ancestor's repaint process will cause
+/// _all_ of its descendent [RenderObject]s to repaint in the absence of any
+/// [RepaintBoundary].
+///
+/// [RepaintBoundary] is therefore used both while propagating the
+/// `markNeedsPaint` flag up the render tree and while traversing down the
+/// render tree via [RenderObject.paintChild] to strategically contain repaints
+/// to the render subtree that visually changed for performance. This is done
+/// because the [RepaintBoundary] widget creates a [RenderObject] that always
+/// has a [Layer], decoupling ancestor render objects from the descendant
+/// render objects.
+///
+/// The upward propagation of the `markNeedsPaint` flag will stop when a
+/// [RepaintBoundary]'s render object is reached. The repainting children can
+/// re-record its display list without re-recording the display list for the
+/// surround tree.
+///
+/// While traversing down, the repainting ancestor can re-use the display list
+/// we recorded previously on descendents behind a [RepaintBoundary], stopping
+/// the repaint traversal beyond that subtree branch.
+///
+/// [RepaintBoundary] has the further side-effect of possibly hinting the engine
+/// to further optimize animation performance when the render subtree behind the
+/// [RepaintBoundary] is sufficiently complex and is static while the surrounding
+/// tree changes frequently. In those cases, the engine may choose to pay a one
+/// time cost of rasterizing and caching the pixel values of the subtree for
+/// faster future GPU re-rendering speed.
+///
+/// The [RepaintBoundary] is also automatically inserted by the framework in
```

Comment:
strike "also" since you haven't mentioned any way that it is automatically inserted yet

Reply:
done